users/vieting/jobs/fairseq_labels.py

```diff
@@ -0,0 +1,252 @@
+    the issue https://github.com/facebookresearch/fairseq/issues/2493 for .wrd and .ltr creation.
+    """
+
+    def __init__(
+        self,
+        audio_dir_path,
+        file_extension="wav",
+        valid_percent=0.01,
+        seed=42,
+        path_must_contain=None
+    ):
+        """
+        :param [tk.Path]|tk.Path audio_dir_path: list of paths or single path to raw audio files to be included
+        :param str file_extension: file extension to look for in audio_dir_path
+        :param float valid_percent: percentage of files to be in validation set
+        :param int seed: random seed for splitting into train and valid set
+        :param str|None path_must_contain: if set, path must contain this substring
+            for a file to be included in the manifest
+        """
+        if isinstance(audio_dir_path, tk.Path):
+            self.audio_dir_paths = [audio_dir_path]
+        else:
+            assert isinstance(audio_dir_path, list)
+            self.audio_dir_paths = audio_dir_path
+        assert min([isinstance(path, tk.Path) for path in self.audio_dir_paths])
+
+        
+        self.manifest_audio_paths = None
+
+        self.file_extension = file_extension
+        self.valid_percent = valid_percent
+        assert 0.0 <= self.valid_percent <= 1.0
+        self.seed = seed
+        self.path_must_contain = path_must_contain
+
+        self.rqmt = {"time": 6, "mem": 8, "cpu": 1}
+
+        self.out_task_path = self.output_path("task", directory=True)
+
+        self.out_train_tsv_path = self.output_path(os.path.join(self.out_task_path.get(), "train.tsv"))
+        self.out_valid_tsv_path = self.output_path(os.path.join(self.out_task_path.get(), "valid.tsv"))
+        
+        self.out_dict_ltr_path = self.output_path(os.path.join(self.out_task_path.get(), "dict.ltr.txt"))
+
+        self.out_train_ltr_path = self.output_path(os.path.join(self.out_task_path.get(), "train.ltr"))
+        self.out_train_wrd_path = self.output_path(os.path.join(self.out_task_path.get(), "train.wrd"))
+        self.out_valid_ltr_path = self.output_path(os.path.join(self.out_task_path.get(), "valid.ltr"))
+        self.out_valid_wrd_path = self.output_path(os.path.join(self.out_task_path.get(), "valid.wrd"))
+        """
+        self.out_valid_tsv_path = self.output_path("valid.tsv")
+        self.out_dict_ltr_path = self.output_path("dict.ltr.txt")
+        self.out_train_ltr_path = self.output_path("train.ltr")
+        self.out_train_wrd_path = self.output_path("train.wrd")
+        self.out_valid_ltr_path = self.output_path("valid.ltr")
+        self.out_valid_wrd_path = self.output_path("valid.wrd")
+        """
+
+    def tasks(self):
+        yield Task("run", rqmt=self.rqmt)
+
+    def run(self):
+        #self.create_manifest()
+        self.create_tsv()
+        self.create_labels(split="train")
+        self.create_labels(split="valid")
+        self.download_dict_ltr()
+
+    def create_tsv(self):
```

Comment:
What is the difference of this and the output of the manifest creation job that we already have?